File: ios/Runner/Api/HostPlayerApiImpl.swift
//
//  HostPlayerApiImpl.swift
//  Runner
//
//  Created by tk on 2024/06/25.
//

import Foundation
import UIKit

class HostPlayerApiImpl: HostPlayerApi {
    func startBySongs(index: Int64) throws {
        let repository = SongRepository()
        let songs = repository.findAll()

        MusicPlayer.shared.start(list: songs, currentIndex: Int(index))
    }
    
    func play() throws {
        
    }
    
    func stop() throws {
        
    }
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/api/HostPlayerApiImpl.kt
package dev.tcode.thinmpf.api

import HostPlayerApi
import android.content.Context
import android.os.Build
import android.util.Log
import androidx.annotation.RequiresApi
import dev.tcode.thinmpf.player.MusicPlayer
import dev.tcode.thinmpf.repository.SongRepository

class HostPlayerApiImpl(private val context: Context): HostPlayerApi {
    override fun startBySongs(index: Long) {
        val repository = SongRepository(context)
        val songs = repository.findAll()
        val musicPlayer = MusicPlayer()

        musicPlayer.start(context, songs, index.toInt())
    }

    override fun play() {
        TODO("Not yet implemented")
    }

    override fun stop() {
        TODO("Not yet implemented")
    }
}